crates/client/rpc/src/lib.rs

```diff
@@ -194,6 +199,29 @@ where
             None
         })
     }
+
+    /// Returns the state diff for the given block.
+    ///
+    /// # Arguments
+    ///
+    /// * `block_hash` - The hash of the block containing the state diff (starknet block).
+    fn get_state_diff(&self, starknet_block_hash: &BlockHash) -> Result<StateDiff, String> {
+        let state_diff = self.backend.da().state_diff(starknet_block_hash).unwrap_or_else(|err| {
+            error!("{err}");
+            ThinStateDiff {
+                deployed_contracts: IndexMap::with_capacity_and_hasher(0, Default::default()),
+                storage_diffs: IndexMap::with_capacity_and_hasher(0, Default::default()),
+                declared_classes: IndexMap::with_capacity_and_hasher(0, Default::default()),
+                deprecated_declared_classes: Vec::with_capacity(0),
+                nonces: IndexMap::with_capacity_and_hasher(0, Default::default()),
+                replaced_classes: IndexMap::with_capacity_and_hasher(0, Default::default()),
```

Comment:
Why can't you just use with_capacity here?
And if it is for a 0 capacity, why not just use `new`?

Reply:
For alignment with what we do in `BlockifierStateAdapter` but agreed it's weird